llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1340,6 +1341,17 @@ RISCVVLOptimizer::checkUsers(const MachineInstr &MI) const {
       continue;
     }
 
+    if (UserMI.isPHI() && PHISeen.insert(&UserMI).second) {
+      LLVM_DEBUG(dbgs() << "    Peeking through uses of PHI\n");
+      for (auto &PhiUse : MRI->use_operands(UserMI.getOperand(0).getReg())) {
+        // If UserMI has a PHI cycle, don't analyze it.
+        if (PhiUse.getParent() == &UserMI)
```

Comment:
Do we need this `PhiUse.getParent() == &UserMI` check here? Does the PHISeen.insert check above already take care of the direct cycle case